### PR TITLE
Adding basic GitHub API calls

### PR DESCRIPTION
**Pull request by peiranli**
The utils.py contains the implementations of the basic GitHub API calls mentioned in the project notes. These functions can be used for collecting the signals.

### CODE REVIEW

File: demo/utils.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        repo_names_by_page = get_repositories_by_page(page, username)
+        if not repo_names_by_page:
+            break
+        repo_names.extend(repo_names_by_page)
+    return repo_names
+
+def get_repositories_by_page(page, username):
+    URL = API + "/users/%s/repos" % (username)
+    query_parameters = {'page': page}
+    response = requests.get(url=URL, params=query_parameters)
```
Comment (jinhongcgoogle) [inline demo/utils.py]: nit:
```
    response = requests.get(url=URL,headers=headers)
    # handle error code and status
    json_response = response.json()
    return json_response
```
seems to be a very common pattern. Sharing error handling code can also increase the debuggability of the code. I think it's worth creating a function for it.

**Review by jinhongcgoogle — COMMENTED** [review]
Overall lgtm.

File: demo/utils.py (hunk 1 above)
Comment (peiranli) [reply demo/utils.py]: Thanks! Will fix it.

**Review by jinhongcgoogle — APPROVED** [review]
lgtm

**Comment by peiranli** [comment]
This demo utils.py code is no longer used.

**Pull request closed without merging**